src/test-metrics.jl

```diff
@@ -38,11 +38,11 @@ function gather_test_metrics(t::AsyncTestSuite)
     if !t.disabled
         for sub_testset in t.sub_testsets
             gather_test_metrics(sub_testset)
-            combine_test_metrics(t, sub_testset)
+            t = combine_test_metrics(t, sub_testset)
         end
 
         for sub_testcase in t.sub_testcases
-            combine_test_metrics(t, sub_testcase)
+            t = combine_test_metrics(t, sub_testcase)
```

Comment:
```suggestion
            combine_test_metrics(t, sub_testcase)
```

This change is not needed, as we always mutate `t::AsyncTestSuite`.

Reply:
look above please.

Reply:
look above please.